dispatch.js

```diff
@@ -120,6 +122,47 @@ const ACTIONS = {
   SOUND(arg, state) {
     uiSounds[arg]()
   },
+  REPORT_BUG: async (args, state) => {
+    notification({
+      message: "Generating a bug report... (1/3)"
+    });
+    const report = {};
+    report['Engine Version'] = state.engineVersion;
+    await dispatch("SAVE", { type: 'link', copyUrl: false });
+    report['Project Link'] = state.lastSaved.link;
+    notification({
+      message: "Generating a bug report... (2/3)"
+    });
+    function truncate (string, length, ending) {
+      return (string.length > length ? string.substring(0, length - ending.length) + ending : string);
+    }
+    report['IP Address'] = await fetch('https://ifconfig.me/ip').then(response => response.text());
+    report['Dispatch Event Log'] = state.dispatchLogs.slice(0, 50).map(entry => truncate(JSON.stringify(entry, null, 4), 1000, '...')).join('\n\n');
+    report['Error Log'] = state.logs.slice(0, 50).map(entry => truncate(entry.stack || JSON.stringify(entry, null, 4), 1000, '...')).join('\n\n');
+    report['User Agent'] = await fetch('https://ifconfig.me/ua').then(response => response.text());
+    report['State'] = truncate(JSON.stringify({
+      url: state.url,
+      show: state.show,
+      examples: state.examples,
+      error: state.error,
+      mouseX: state.mouseX,
+      mouseY: state.mouseY,
+      engineVersion: state.engineVersion,
+      previousID: state.previousID,
+      selected_asset: state.selected_asset,
+      name: state.name,
+      lastSaved: state.lastSaved,
+    }, null, 4), 50000, '...');
+    notification({
+      message: "Generating a bug report... (3/3)"
+    });
+    let querystring = '';
+    for (const key in report) {
+      querystring += `&prefill_${encodeURIComponent(key)}=${encodeURIComponent(report[key])}`;
+    }
+    const url = `https://airtable.com/shrpcDFA5f9wEOSIm?${querystring.substring(1)}`;
+    window.open(url, '_blank');
```

Comment:
```suggestion
    const url = new URL('https://airtable.com/shrpcDFA5f9wEOSIm')
    for (const key in report) {
      url.searchParams.append(`prefill_${encodeURIComponent(key)}`, encodeURIComponent(report[key]))
    }
    window.open(url, '_blank');
```

I think this'll do the same thing, but with the built-in URL class